[GLib] Bump font dependencies

mcatanzaro opened the pull request:
#### 8701fcdb29a3e71dbddbd345e7b49c42432a7854
<pre>
[GLib] Bump font dependencies
<a href="https://bugs.webkit.org/show_bug.cgi?id=252682">https://bugs.webkit.org/show_bug.cgi?id=252682</a>

Reviewed by Carlos Garcia Campos.

Also, clean up some USE(HARFBUZZ) and ENABLE(OPENTYPE_MATH) build
guards.

 * Source/WebCore/platform/graphics/FontPlatformData.h:
 * Source/WebCore/platform/graphics/freetype/FontPlatformDataFreeType.cpp:
 * Source/WebCore/platform/graphics/opentype/OpenTypeMathData.h:
 * Source/cmake/OptionsGTK.cmake:
 * Source/cmake/OptionsWPE.cmake:

Canonical link: <a href="https://commits.webkit.org/260744@main">https://commits.webkit.org/260744@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3cbed11462a1376f9dcaf387a668a77f0934056e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109062 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18144 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41884 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/597 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118329 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/112947 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19603 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9446 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101302 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114822 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14699 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97937 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42869 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96671 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29590 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84603 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/98092 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10926 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30936 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/98884 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/9018 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11665 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7857 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31022 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17040 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50535 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/106558 "Built successfully") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7415 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13271 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26410 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/cmake/OptionsGTK.cmake:
```diff
@@ -45,12 +45,6 @@ if (WTF_CPU_ARM OR WTF_CPU_MIPS)
     SET_AND_EXPOSE_TO_BUILD(USE_CAPSTONE ${DEVELOPER_MODE})
 endif ()
 
-# For old versions of HarfBuzz that do not expose an API for the OpenType MATH
-# table, we enable our own code to parse that table.
-if ("${PC_HARFBUZZ_VERSION}" VERSION_LESS "1.3.3")
-    add_definitions(-DENABLE_OPENTYPE_MATH=1)
-endif ()
```
This means we must ensure `OPENTYPE_MATH` is always disabled. See PlatformEnable.h:

```
#if !defined(ENABLE_OPENTYPE_MATH) && (OS(DARWIN) && USE(CG)) || (USE(FREETYPE) && !PLATFORM(GTK)) || (PLATFORM(WIN) && (USE(CG) || USE(CAIRO)))
#define ENABLE_OPENTYPE_MATH 1
#endif
```
That means we would enable it for WPE, so I think we should remove the `(USE(FREETYPE) && !PLATFORM(GTK))`

We should also change `#if USE(HARFBUZZ) && !ENABLE(OPENTYPE_MATH)` with `#if USE(HARFBUZZ)`

carlosgcampos replied:
I've seen PR 10461 now, we still need to change the #ifdefs, though

mcatanzaro replied:
Updated.

mcatanzaro replied:
Do you want me to remove these guards too?

```
WebCore/platform/graphics/freetype/FontCacheFreeType.cpp:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/freetype/FontCacheFreeType.cpp:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/freetype/FontCacheFreeType.cpp:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/freetype/FontCacheFreeType.cpp:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/freetype/FontCacheFreeType.cpp:#endif // ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/freetype/FontCacheFreeType.h:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/freetype/FontCustomPlatformDataFreeType.cpp:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/freetype/FontCustomPlatformDataFreeType.cpp:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/freetype/FontCustomPlatformDataFreeType.cpp:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/freetype/FontPlatformDataFreeType.cpp:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/freetype/FontPlatformDataFreeType.cpp:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/harfbuzz/ComplexTextControllerHarfBuzz.cpp:#if ENABLE(VARIATION_FONTS)
WebCore/platform/graphics/harfbuzz/ComplexTextControllerHarfBuzz.cpp:#if ENABLE(VARIATION_FONTS)
```

I have left them for now.

carlosgcampos replied:
I guess we could remove them, I think we have other cases of ifdefs or feature we unconditionally enable, so it's up to you.

mcatanzaro committed:
[GLib] Bump font dependencies
https://bugs.webkit.org/show_bug.cgi?id=252682

Reviewed by Carlos Garcia Campos.

Also, clean up some USE(HARFBUZZ) and ENABLE(OPENTYPE_MATH) build
guards.

 * Source/WebCore/platform/graphics/FontPlatformData.h:
 * Source/WebCore/platform/graphics/freetype/FontPlatformDataFreeType.cpp:
 * Source/WebCore/platform/graphics/opentype/OpenTypeMathData.h:
 * Source/cmake/OptionsGTK.cmake:
 * Source/cmake/OptionsWPE.cmake:

Canonical link: https://commits.webkit.org/260744@main
